docs/content/sips/006-spin-plugins.md

```diff
@@ -0,0 +1,233 @@
+                    "sha256"
+                ],
+                "properties": {
+                    "os": {
+                        "type": "string",
+                        "enum": [
+                            "linux",
+                            "osx",
+                            "windows"
+                        ]
+                    },
+                    "arch": {
+                        "type": "string",
+                        "enum": [
+                            "amd64",
+                            "aarch64"
+                        ]
+                    },
+                    "url": {
+                        "type": "string"
+                    },
+                    "sha256": {
+                        "type": "string"
+                    }
+                },
+                "additionalProperties": false
+            }
+        }
+    },
+    "additionalProperties": false
+}
+```
+
+A plugin manifest defines a plugin’s name, version, license, homepage (i.e. GitHub repo), compatible Spin version, and gives a short description of the plugin. It also points to the plugin source for various operating systems and platforms.
+
+The `name` and `spinCompatibility` fields have specific format conventions.
+
+#### Spin Plugin Naming Conventions
+
+The following naming conventions are to be followed for plugins where `$name` is the name of the plugin.
+
+- The `name` field in the plugin manifest must be `$name`.
+- Even if the majority of plugins live within the Spin plugins repository, there is a need to distinguish between plugins that are maintained by Spin vs community plugins. They will be distinguished via the plugin name inside the manifest. The name of community plugins must not have "spin" as a prefix, while plugins maintained by Spin should contain a prefix of `spin-`.
+- Manifests for older versions of the plugin can be retained in the Spin Plugins repository named `$name@$version.json` where `$version` is the value of the `version` field of the manifest. These specific versions can be installed using the `--version` flag.
+- The binary of the plugin must be named `$name`
+- The latest plugin manifest file must be named `$name.json`
+- The license for the plugin must be named `$name.license`
+
+#### Plugin Compatibility
+
+Spin plugins must specify compatible versions of Spin in the `spinCompatibility` field of the manifest. The field is expected to be a [comparison operator](https://docs.rs/semver/1.0.13/semver/enum.Op.html) (`=, >, >=, <, <=, ~, ^, *`) along with the compatible version of Spin. The JSON schema validates that the `spinCompatibility` field is a string that matches the following regular expression: `^[><~^*]?[=]?v?(0|[1-9]\d*)(\.(0|[1-9]\d*))?(\.(0|[1-9]\d*))?(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$`. 
+For example, specifying `=0.4` means that the plugin is compatible with versions equivalent to `>=0.4.0, <0.5.0`.
```

Comment:
Do we want to accept multiple rules to the field? For example setuptools allows the form `Django>=1.4.2,<1.8.99,!=1.5,!=1.6` which means

- install Django
- ALLOW `>=1.4.2, <= 1.8.99`
- EXCEPT `>=1.5.0, <1.7.0`

Which could be useful if it is incompatible with few specific versions of spin.

Reply:
added in 1cd903d28425e0